tilelang/tools/bench.py

```
@@ -0,0 +1,70 @@
+                f"{func.__module__}.{func.__name__}  {avg_latency:.2f} ms  (fail {fail_count}/{repeat})"
+            )
+    else:
+        print(f"{func.__module__}.{func.__name__}  FAILED (no valid run)")
+
+
+def main():
+    # Entry point — automatically run all bench_* functions in caller file.
+    test_file = inspect.getsourcefile(sys._getframe(1))
+    module = {}
+    with open(test_file) as f:
+        exec(f.read(), module)
```

Ruff (0.14.4)

Use of `exec` detected

(S102)